packages/xstate-test/test/coverage.test.ts

```diff
@@ -0,0 +1,389 @@
+    const feedbackMachine = createTestMachine({
+      id: 'feedback',
+      initial: 'logon',
+      states: {
+        logon: {
+          initial: 'empty',
+          states: {
+            empty: {
+              on: {
+                ENTER_LOGON: 'filled'
+              }
+            },
+            filled: { type: 'final' }
+          },
+          on: {
+            LOGON_SUBMIT: 'ordermenu'
+          }
+        },
+        ordermenu: {
+          type: 'final'
+        }
+      }
+    });
+
+    const model = createTestModel(feedbackMachine);
+
+    for (const plan of model.getShortestPlans()) {
+      await model.testPlan(plan);
+    }
+
+    const coverage = model.getCoverage(coversAllStates());
+
+    expect(coverage).toHaveLength(5);
+
+    expect(coverage.filter((c) => c.status !== 'covered')).toHaveLength(0);
+
+    expect(() => model.testCoverage(coversAllStates())).not.toThrow();
+  });
+
+  it('skips filtered states (filter option)', async () => {
+    const TestBug = createTestMachine({
+      id: 'testbug',
+      initial: 'idle',
+      context: {
+        retries: 0
+      },
+      states: {
+        idle: {
+          on: {
+            START: 'passthrough'
+          }
+        },
+        passthrough: {
+          always: 'end'
+        },
+        end: {
+          type: 'final'
+        }
+      }
+    });
+
+    const testModel = createTestModel(TestBug);
+
+    const testPlans = testModel.getShortestPlans();
+
+    const promises: any[] = [];
+    testPlans.forEach((plan) => {
+      plan.paths.forEach(() => {
+        promises.push(testModel.testPlan(plan));
+      });
+    });
+
+    await Promise.all(promises);
+
+    expect(() => {
+      testModel.testCoverage(
+        coversAllStates({
+          filter: (stateNode) => {
+            return stateNode.key !== 'passthrough';
+          }
+        })
+      );
+    }).not.toThrow();
+  });
+
+  // https://github.com/statelyai/xstate/issues/981
+  it.skip('skips transient states (type: final)', async () => {
+    const machine = createTestMachine({
+      id: 'menu',
+      initial: 'initial',
+      states: {
+        initial: {
+          initial: 'inner1',
+
+          states: {
+            inner1: {
+              on: {
+                INNER2: 'inner2'
+              }
+            },
+
+            inner2: {
+              on: {
+                DONE: 'done'
+              }
+            },
+
+            done: {
+              type: 'final'
+            }
+          },
+
+          onDone: 'later'
+        },
+
+        later: {}
+      }
+    });
+
+    const model = createTestModel(machine);
+    const shortestPlans = model.getShortestPlans();
+
+    for (const plan of shortestPlans) {
+      await model.testPlan(plan);
+    }
+
+    // TODO: determine how to handle missing coverage for transient states,
+    // which arguably should not be counted towards coverage, as the app is never in
+    // a transient state for any length of time
```

Comment:
isn't it still important for coverage to check if a transient state was visited though?

Reply:
Good question; I don't think so. For MBT, a transient state is used as a mechanism for going to non-transient states, not as an actual state to be tested, so its coverage doesn't matter.

If a "transient" state is somehow represented in the SUT, then 1) it should be represented as a real state, not as a transient state, and 2) it's not really a transient state anyway; there's a non-zero amount of time the SUT will be in that state.

Reply:
hm, i guess this somewhat depends on how do we define a transient state - is a `{ type: 'final' }` without the associated `onDone` considered to be transient?

Reply:
No, I don't think so.

Reply:
We'll leave this comment in here and revisit it if someone reports it.

It was only an issue when test coverage was yelling that it was missing, but we're removing coverage testing.